### ISSUE TITLE
Fantastic piece of sofware!

### ISSUE THREAD

**Issue by James-F2** [ledoge/novideo_srgb#18]
This allowed me to use all the options in my monitor that are typically locked in sRGB Picture Mode.
The Dithering option eliminated all banding issues I saw in v1.0, and the custom gamma curve is great.

Even though I have an i1 Display Pro, for simplicity I create a "Synthetic ICC" in DisplayCAL with my EDID xy coordinates and custom gamma curve so I can tweak it further in Novideo sRGB, or I can measure my actual xy coordinates with the i1 and then create a Synthetic ICC.
Now I have control over my monitor without being limited by the sRGB Picture Mode that does not allow me to change anything, AND have accurate REC.709 gamut in games, browsing, etc.

Thank you again for this amazing software. 

PS.
An icon for the software will be great, it looks like a virus without an icon. haha

**Comment by James-F2** [comment]
Another thing that would help me update the software is an "About" button with a link to this Github page, and remind me that LeDoge created this awesome tool.

**Comment by ledoge** [comment]
Thank you for the feedback! An "About" window seems like a good idea – I'll add that in the next release. Also, an icon would indeed be nice to have, but I lack the artistic ability to make a decent one...

**Comment by James-F2** [comment]
Same icon as DWM LUT, just so it doesn't look suspicious.

**Comment by James-F2** [comment]
@ledoge 
If I do a full greyscale calibration in DisplayCAL, will novideo_srgb take all 256 steps from the ICC for the gamma curve calculation or will it only use the 0,0,0 black point?
In other words, will it 'fix' a poorly behaved display that clips several black shades?

**Comment by ledoge** [comment]
The tone response curves are used in two ways: the black point of the display is calculated using them, and the re-gamma LUT is populated with the inverse of the curves. So I think there shouldn't be any issues?

**Comment by James-F2** [comment]
No issue, I just wanted to know if there is any benefit of Calibrating+Profiling or just Profiling is enough to get the black value.
I'll rephrase, is the calculated gamma curve using ONLY the black point, or is it using all measured greyscale values?

**Comment by ledoge** [comment]
>if there is any benefit of Calibrating+Profiling or just Profiling is enough to get the black value

Calibrating just adds vcgt data to the profile (which might result in a more accurate profile?), but novideo_srgb works perfectly fine with or without vcgt data. It's not relevant for the calculation of the black point.

>is the calculated gamma curve using ONLY the black point, or is it using all measured greyscale values

There are two stages in the color side space conversion pipeline that affect gamma: The first 1D LUT is used to convert the pixel values to linear light. Its values are calculated only from the selected gamma (e.g. sRGB) and the black point. The second 1D LUT is used to convert from linear light to the right values for the display - this is where all of the measured grayscale values and, if present, the vcgt data come in.

**Comment by James-F2** [comment]
I see, so there is a benefit in Profiling with a large amount of measured greyscale patches.
Thank you!

**Comment by James-F2** [comment]
One last thing for people who are using DisplayCAL and a calibration device.
If you are using DisplayCAL to create an ICC make sure you select "XYZ LUT+Matrix" in the Profiling tab because Matrix only profiles will not produce a correct gamma curve in novideo_srgb.
I use a custom Testchart with 256 grey patches (neutral patches) for maximum greyscale accuracy, and at least 400 color patches.
Also, there is no need for Calibration, so leave everything there in "As Measured" like you would with MadVR calibration, except Interactive white point adjustment so you can bring your display closer to 6500K manually with R,G,B controls in your monitor OSD before Profiling.

I've attached a REC.709 Calibration Verification with the ICC applied in novideo_srgb, my monitor is in Standard wide gamut mode.
Mind blowing results that compare to MadVR with its advanced 16bit processing and dithering, and it runs natively on the  Nvidia GPU. 
Novideo_srgb deserves a lot more recognition.

[Measurement Report 3.8.9.3 — Gigabyte M32Q @ 0, 0, 2560x1440 — 2022-03-08 13-19.html.zip](https://github.com/ledoge/novideo_srgb/files/8205695/Measurement.Report.3.8.9.3.Gigabyte.M32Q.%40.0.0.2560x1440.2022-03-08.13-19.html.zip)

**Comment by ledoge** [comment]
Wow, those results are awesome! Thanks for sharing.

>Matrix only profiles will not produce a correct gamma curve in novideo_srgb

They should work fine as long as you disable black point compensation.

**Comment by James-F2** [comment]
Yeah, amazing that this level of accuracy can be applied natively at the GPU level (including advanced dithering), without the need of external hardware, expensive grading monitors, or specialized software.

It is my greatest pleasure using your software and software by people like Graeme Gill and Florian Höch.
I am very grateful for this.

James

**Comment by aufkrawall** [comment]
@James-F2 Would you mind sharing your testchart for DisplayCal? :)

**Comment by James-F2** [comment]
Not at all.

It's nothing special.
256 neutral patches, and 7 for Single and Multidimensional color patches.
I think 7 is way to much, 5 is more than enough, but I always do 256 neutral patches so there is minimal interpolation, especially in the dark region.

[256greyscale testchart.zip](https://github.com/ledoge/novideo_srgb/files/8209304/256greyscale.testchart.zip)

![testchart](https://user-images.githubusercontent.com/31185814/157317497-ad05d881-b580-4686-92fb-7d9e4a1463e8.png)

**Comment by aufkrawall** [comment]
Thanks, I'll give it a try and report back!

**Comment by markanini** [comment]
Curious if custom testcharts improve verification scores in this application. Before using novideo_rgb on my consumer grade WLED display I got the best scores with the default 175 patch set, somewhere on DisplayCAl forums Florian mentioned that testchart had some special optimization, I don't remember the details. 

> I create a "Synthetic ICC" in DisplayCAL with my EDID xy coordinates 
How do you gather those from an existing profile?

**Comment by ledoge** [comment]
>Curious if custom testcharts improve verification scores in this application.

I don't really know much about how ArgyllCMS works internally, but what I'd expect is that if you're generating an XYZ LUT profile, adding more color patches won't help but adding grayscale ones will (since novideo_srgb samples the extreme points and the grayscale axis only). With a 3 curves + matrix profile (make sure to disable black point compensation!) I think adding more grayscale and color patches should improve the average delta e.

**Comment by James-F2** [comment]
> since novideo_srgb samples the **extreme points and the grayscale axis only**

Ah, thank you, that clarified a lot.
So there is no need for color patches at all, only greyscale and three 100% RGB values, I see.

Thanks!

**Comment by aufkrawall** [comment]
> I skip calibration and Profile only, it provides enough data and is much faster since there is no iterative calibration to minimize dE during measurement.

Yeah, maybe my Spyder 4 is too shitty and is more dependent on those extensive calibration tests. Just a wild guess though. novideo_srgb clamp result with default 1x curve chart + medium speed calibration probably still wipes the floor with almost any monitor internal sRGB clamp:
[Measurement Report 3.8.9.3 — G27q-20 @ 0, 0, 2560x1440 — 2022-03-08 22-57.zip](https://github.com/ledoge/novideo_srgb/files/8225822/Measurement.Report.3.8.9.3.G27q-20.%40.0.0.2560x1440.2022-03-08.22-57.zip)

Of course a driver/OS API for full blown 3D LUT correction would be the nonplusultra. I guess you'd find higher deviations when just clamping the gamut volume when running a really extensive validation test chart. But that's probably hard/impossible to notice during everyday usage, depending on the display's linearity.

**Comment by James-F2** [comment]
I have to say that the ease of jumping between calibrated and accurate gamma curves without creating several different 3D LUTs for MadVR is absolutely delightful.
I use sRGB, BT.1886 and Relative 2.35+0% Offset curves, and it has never been this easy with this level of accuracy before.

My collection of calibration test images confirm that everything is working perfectly and the resulting gamma curves are accurate, smooth, and without black crush.
[Calibration Pictures.zip](https://github.com/ledoge/novideo_srgb/files/8234567/Calibration.Pictures.zip)
For me, the key was 256 neutral patches (all 8bit steps) with Curves+Matrix profiling.

I'm still mind blown by how accurate, easy, and bug free this is, can't bother with 3D LUTs anymore.

**Comment by erusyns** [comment]
hey james, do you mind making a make a step by step on how you did this, Im having hard time understanding the process. I have a calibrated before using XYZ Lut + Matrix.

**Comment by James-F2** [comment]
Here's a summary;

Skip calibration by setting everything to 'as measured' only do 'interactive display adjustment' for 6500K white point.
Profile Curves+Matrix (black point compensation disabled) with 256 neutral patches and 2 single channel (color) patches or use this test chart.
[256 Neutral Testchart.zip](https://github.com/ledoge/novideo_srgb/files/8235126/256.Neutral.Testchart.zip)

For verification I use these settings, with novideo_srgb clamped to sRGB;
![verification](https://user-images.githubusercontent.com/31185814/157949525-d57772ec-3b50-4e3c-a8f8-7bab0774f481.png)

**Comment by James-F2** [comment]
I am not sure how novideo_srgb is using color patches from the Matrix profile though, and whether there is a benefit of using more than just three 100% primary RGB patches.

**Comment by erusyns** [comment]
thank you james

**Comment by ledoge** [comment]
>I am not sure how novideo_srgb is using color patches from the Matrix profile though, and whether there is a benefit of using more than just three 100% primary RGB patches.

AFAICT ArgyllCMS will use the color patches to generate an optimized matrix, so there might be a slight benefit for average delta Es at least.

Also, I made some experimental changes that should (assuming I did it properly) result in slightly better 8-bit gamma tracking. Please try [this](https://github.com/ledoge/novideo_srgb/files/8236476/novideo_srgb.zip) build and see if you get lower grayscale errors compared to the current version. The differences might be very minor though, so I'd suggest comparing it to a regular DisplayCAL verification without "Use simulation profile as display profile" as well to make sure the ICC profile is still accurate (or make a new one).

**Comment by James-F2** [comment]
I Profiled a new ICC with 256 neutral and 3x25 color patches.
Leaving the i1 Display Pro on screen after profiling, I see no difference in verification between release and test build, I made sure to refresh the "Clamped" button between .exe swaps.

**Comment by markanini** [comment]
Depending on the profile with the test version average Delta E dropped .1 - .05. 
I made the profiles last night.

Independently I compared the average delta E for the different profile types in DisplayCal:  XYZLUT+matrix, Curves+matrix
Single curve+matrix, Gamma+matrix, Single gamma+matrix. Going down the list the average Delta E increases, but the first two are very close. So far the test used Tone curve:As measured.

Setting Tone Curve to Gamma 2.2 and going from Calibration Speed:Vary fast to Slow the average Delta E increases! So my chosen profile uses Tone curve:As measured and Profile type:XYZLUT+matrix for the lowest Average delta E. My measurment device is a i1Display Pro.

I ran out of time to make a comparison were DisplayCal loads the profile instead, or using James-F2's test chart so one of you guys can help answer that one.

**Comment by James-F2** [comment]
I see an improvement in the smoothness of the greyscale with v2.7.
There was a small 'kink' in the "16Bit Gradient.png" test pattern (I shared it above) with 2.6 and the test build (dithering enabled), now it's gone with 2.7 and greyscale is smooth as eggs.

**Comment by ledoge** [comment]
Very interesting – I thought that test build should have perfectly accurate grayscale, and with v2.7 I removed part of that to get better color accuracy with slightly worse grayscale accuracy, so I'm not sure why it would be even better now, but it's very nice that it is!

**Comment by James-F2** [comment]
Maybe it was a rounding error before 2.7?
My monitor is capable of 10bit gradation but I always work in 8bit because it is the most compatible with software, particularly when switching between OpenGL and Direct3D, 8bit is still the standard.

Here are some images of a Grey Ramp with the Test Build and v2.7, notice the kink doesn't smooth out even with 8bit Dithering On with the Test Build.
v2.7 with Dithering On is perfectly smooth, as without Clamping,

Test Build Dithering OFF
![Test Build Dithering Off](https://user-images.githubusercontent.com/31185814/158051917-848e8397-b8b2-4711-a185-3ce8100c78ca.jpg)

Test Build Dithering ON, a faint bright kink is still visible.
![Test Build Dithering On](https://user-images.githubusercontent.com/31185814/158051921-d264a72b-413c-4b3d-a894-0159108b4511.jpg)

v2.7 Dithering OFF
![2 7 Dithering Off](https://user-images.githubusercontent.com/31185814/158052349-96a6751f-3392-4dcc-88a3-7285c8851f81.jpg)

v2.7 Dithering ON, perfectly smooth in person, no repetition of 8bit grey lines, and new gamma tracking is spot on.
![2 7 Dithering On](https://user-images.githubusercontent.com/31185814/158052350-e0faeee5-1a42-4427-b3c1-73c273e2ca4d.jpg)

**Comment by ledoge** [comment]
>When you say "Optimize de-gamma LUT for 8 bit values" what it was optimized for before that?

Before that, I just filled the de-gamma/re-gamma LUTs "normally", which meant that 10 bit values would be mapped exactly, but 8 bit values would not as they end up being interpolated between two entries in the de-gamma LUT. The optimization I did is identifying the entries that would be used for interpolation and putting the correct values for 8 bit inputs in them. Since there are pairs of entries with identical values now, the bit depth is effectively reduced to 9 bits, so there's probably a bit of banding with 10 bit inputs.

**Comment by James-F2** [comment]
For the record, when I switch to 10bit in nvidia control panel and use 10bit dithering, the bright kink is still present in v2.6 and the test build.

Shouldn't 8bit values be aligned in 10bit by default without interpolation?

**Comment by ledoge** [comment]
>For the record, when I switch to 10bit in nvidia control panel and use 10bit dithering, the bright kink is still present in v2.6 and the test build.

From my understanding, since the desktop is still being rendered in 8 bits, the values going into (and coming out of) the color space conversion pipeline are the same, so this is expected. The only difference is that the values get converted to 10 instead of 8 bits before being sent to the display.

>Shouldn't 8bit values be aligned in 10bit by default without interpolation?

That does seem like it should work, but unfortunately it doesn't. 1023/255 is 4.01176.... You're only getting nicely aligned conversions if you go all the way up to 16 bits, as 65535/255 = 257 (IIRC the Windows VCGT loader produces slightly wrong results because it assumes that this number is 256).

**Comment by James-F2** [comment]
Interesting, I thought it should be 1024/256, zero included.
When novideo_srgb is disabled I don't see any change in greyscale when my display is set to 8 or 10 bit, the input is of course always 8bit from windows.
Maybe nvidia driver is doing some kind of conversion form 8bit input to 10bit with dithering when the display is set to 10bit?

8bit is still the standard for most if not all non-color-aware applications and rounding to 8bit was a great decision.
If full 10bit pipeline is still desirable for the rare applications that outputs real 0-1023 values (like madVR in exclusive mode or Photoshop) and bypasses DWM, maybe you can make it switchable?

In any case, I believe 8bit should be the default and 10bit processing can be a switchable option if needed.
Warning though, people will enable 10bit even if it makes grey scale less accurate for 8bit input, same with dithering, because 10 is more than 8... DERP.
The grey ramp test pattern helps a lot to visually verify things.

v2.7 made Novideo_sRGB more accurate than it ever was.

**Comment by ledoge** [comment]
>Interesting, I thought it should be 1024/256, zero included.

Yeah, it's kinda counterintuitive. Try thinking about it like this: the 8-bit value `0` has to be mapped to the 10-bit value `0` (the minimum value), and the 8-bit value `255` has to be mapped to the 10-bit value `1023` (the maximum value). The only (linear) operation that satisfies these properties is multiplication by `1023/255`.

>Maybe nvidia driver is doing some kind of conversion form 8bit input to 10bit with dithering when the display is set to 10bit?

I'm not sure, but it would probably make more sense than truncating to the nearest 10 bit value.

>In any case, I believe 8bit should be the default and 10bit processing can be a switchable option if needed.
Warning though, people will enable 10bit even if it makes grey scale less accurate for 8bit input, same with dithering, because 10 is more than 8... DERP.

I definitely agree. It's certainly a niche use case, but I'll try to think of a good label so that I can add it as an option without everyone enabling it when they shouldn't.

**Comment by ledoge** [comment]
Just pushed a release with that option added – I think the label is ambiguous and "scary" enough that people won't just enable it randomly.

**Comment by James-F2** [comment]
Testing it now.

I'm not sure if the 10bit switch is working, I don't see a change from 8bit.

**Comment by ledoge** [comment]
Weird, I double checked using the debugger and the switch is definitely working as intended.

**Comment by James-F2** [comment]
Could be, I'm just not seeing the same kink as in the 2.6 or test build.

**Comment by James-F2** [comment]
Wait... I'm sorry, I do see a change but it is very tiny.
The change is extremely small grey hue shift in the kinks when dithering is disabled, barely visible unless zoomed in 2000% on the kink.

**Comment by ledoge** [comment]
I have absolutely no idea why there'd be a kink with v2.6 but not with v2.8 and the optimization disabled, as the math should be 100% the same... Good to know that it's at least doing *something* though.

**Comment by James-F2** [comment]
Yeah, the math is exactly the same, except now it samples a ready variable instead of an equation.
Maybe the compiler prefers filling arrays form ready variables instead of calculating them on the spot... timing?... just guessing.

Anyway, I'd like to think that the "new" 10bit mode in 2.8 is more accurate, because I do not think displaying 10bit in 8bit should have that clearly visible errors as I saw in 2.6, I think it is a very welcome chance fix.

**Comment by aufkrawall** [comment]
Can confirm that the new 8 bit optimization helps. It's very noticeable with a black - orange 8 bit gradient when dithering is disabled. And also with temporal dither to 8 bit, there is still a bit of an improvement in the darker shades. Nice! :)

**Comment by ledoge** [comment]
@James-F2 I compared v2.6 vs v2.8 with the optimization disabled by using a separate program to dump the LUT contents, and they're 100% identical. Are you sure you were using the same settings and the same profile when comparing them?

**Comment by James-F2** [comment]
Not in 2.6, but the Test Build you posted in this thread.
This post: https://github.com/ledoge/novideo_srgb/issues/18#issuecomment-1065651500

With this Test Build I see a bright kink, is it not identical to 2.6?

**Comment by aufkrawall** [comment]
@ledoge I suppose this is a driver bug, but did you notice that active clamp makes gamma of the hardware mouse cursor too bright? This is hard to notice with the Windows default cursor theme, but when e.g. games make use of a custom hardware cursor theme, this is more distinct.

**Comment by ledoge** [comment]
Interesting, haven't noticed that myself, but I did notice what I believe to be a similar issue with HDR. If you're in SDR mode and have an HDR application (e.g. the DisplayHDR Test app) fullscreened, gamma is way too dark.

Is your cursor entirely blown out or just a bit too bright? Would be interesting to test if this also happens when applying a novideo_srgb calibration using something like the sRGB profile, as that may put the regamma LUT in a similar state as with the color space conversion pipeline not being used.

**Comment by aufkrawall** [comment]
It's quite substantially brighter, yes. It also happens with sRGB gamma curve. It seems to happen after dragging a window a bit (which temporarily activates software cursor until dropping the window), but moving the cursor over resize borders can reset it back to normal.

**Comment by James-F2** [comment]
I currently favor Relative Gamma 2.2 with 0% Black Output Offset, this is the closest an LCD can get to pure 2.2 gamma without crushing black shades, I think it looks better than sRGB curve for everything, including Movies, Games, and Internet.

Same with settings above for Gamma 2.4 for Movies, which I prefer over BT.1886 that elevates black shades like sRGB.
In madVR you can increase gamma to 2.4 with the same results if you starting with Relative 2.2 0% in novideo_srgb.

As for Dithering, I'm using SpatialDynamic2x2, after careful comparison of all options in 6bit I found this gives the smoothest results with the least amount of noise and no visible dithering pattern.

A tip regarding calibration;
Plug the measuring device USB and let it warm up on the table from 30 minutes, only put it on the screen just before calibrating, take it off the screen after calibrating. In other words don't let it warm up or receive constant light.
I find it results in a better calibration more accurate gamma tracking.

**Comment by aufkrawall** [comment]
Did some more testing: I think configuring more than 2 single channel patches (i.e. 1x R,G,B in actual patches) doesn't seem to help for lower delta values. I ran calibration with just that and 256 neutral patches and got my best result so far with 3x curves (Maximum ΔE*00 0.71, maximum gamma deviation 0.06).

Curiously, running that profiling config with a different system (different display and colorimeter) again lead to a color tint with very dark gray when applying calibration via novideo_srgb (gets more visible with e.g. 0% black output offset). I intend to check what happens with less grayscale patches there. I'm getting this also with my system when using more single channel patches, so what I can say already is that it seems to be connected to the number of patches used with 3x curves.

I'd be curious if 3x curves are mathematically expected to be significantly better than 1x?

**Comment by James-F2** [comment]
The calibration is as accurate as your meter with different spectral distribution of the panel backlight.

I get spectacular results with no visible tint or coloration down to 0 black, with 3x Curves + Matrix (256 grey and 2 single channel patches), i1 Display Pro.

X-Rite rebranded the i1 Display series as "Calibrite ColorChecker Display Pro" now.

**Comment by aufkrawall** [comment]
> The calibration is as accurate as your meter with different spectral distribution of the panel backlight.

Well, there never is any close to black tint also with 3x curves when calibration is applied via VCGT/gamma ramps instead. I'll have another look what happens with such an "affected" profile when generating a 3D LUT from it and applying it via ledoge's other great tool dwm_lut. My layman gut feeling is that under some circumstances something goes wrong with close to blacks in novideo_srgb and 3x curves profile. When a 3D LUT generated from the same profile looks fine, I think it can be assumed that nothing's wrong with the measurements or the profile per se.

**Comment by ledoge** [comment]
>Curiously, running that profiling config with a different system (different display and colorimeter) again lead to a color tint with very dark gray when applying calibration via novideo_srgb

Not really sure why that would happen, I'd assume that such a tint could be caused by generating a matrix optimized for many color patches, but that's not the case here. Maybe try generating an XYZ LUT profile from the same measurement data and see if that works better?

>I'd be curious if 3x curves are mathematically expected to be significantly better than 1x?

It really depends on the display, as you need 3 curves to describe its behaviour unless the 3 channels can all be described by the same tone response curve. So either the monitor needs to have near-identical response curves on each channel natively, or you can use a VCGT calibration to "align" them while targeting some gamma. With no VCGT calibration, I'm pretty sure you should always do 3 curves.

**Comment by markanini** [comment]
You can to test 1x curve without profiling again with file-create profile from measurement data.

I prefer the 3xcurves+Matrix and the default matrix test chart containing 29 neutral patches. It's stays a little closer to target gamma near black. Comparing to a test chart with 256 neutral patches average DeltaE is higher but not a huge difference. I always set tone curve (and white point) to as measured.

**Comment by James-F2** [comment]
DeltaE of the black shades is dependent on the Tone Curve selected in the Verification tab.
I verify with Relative 2.2. 0% Offset.

**Comment by markanini** [comment]
Good point, use them too. However most test ramps are not in a perceptual color space and cause an inflated chase for near black shade differentiation. The lagom.nl black test is also an offender.

**Comment by aufkrawall** [comment]
Again regarding that dark grayscale tint issue: I thought it was the same issue with the VA device as it was with my IPS device when using more color patches with 3x curves + matrix. But it turned out it was not, also single curve + matrix could exhibit it, also with just 3 color patches and relatively few grayscale patches. It also wasn't isolated to novideo_srgb, the dark gray color tint also occured with 3D LUT applied via dwm_lut.

So for whatever weird reason, that VA display requires to also perform the calibration step with DisplayCal, then dark gray doesn't have weird color tints. We used XYZ LUT + matrix format profile with default test chart. We didn't try a new calibration in 3x curves + matrix format yet, as it takes forever with the Spyder 4 and calibration set to medium. I suppose gamma curve result is a bit worse that way vs. preferred 3x curves without calibration step, but it's still well within an acceptable margin.

**Comment by erusyns** [comment]
hey james how do get 3x curves + matrix? i only have option for curves+matrix

**Comment by James-F2** [comment]
"Curve**s**" is 3 curves (R,G,B).
The other is "Single Curve" which is r,g,b, locked to the same value.

VA panels are very slow in dark shades, this is most probably the reason aufkrawall see some tint there without calibration, especially with temporal dithering.

**Comment by erusyns** [comment]
> "Curve**s**" is 3 curves (R,G,B). The other is "Single Curve" which is r,g,b, locked to the same value.
> 
> VA panels are very slow in dark shades, this is most probably the reason aufkrawall see some tint there without calibration, especially with temporal dithering.

thank you, im struggling to get my black and red to <1. can you check my step and possibly point out what i did wrong?

https://drive.google.com/file/d/11m-hZi03f5JE-_DcN-4WH0TV0HgbPfe2/view?usp=sharing

**Comment by James-F2** [comment]
Your results are excellent, don't obsess.

**Comment by erusyns** [comment]
> Your results are excellent, don't obsess.

lol, i cant really stand the red popping out from other colours.

**Comment by aufkrawall** [comment]
> lol, i cant really stand the red popping out from other colours.

Well, according to you results, it shouldn't. At least not more than 100% red usually does as a signal color to human perception. But why did you set brightness to 55% in the Nvidia control panel? Increase backlight brightness instead if you find the result too dark.

You can't "fix" the deviation for black, I suspect it's the color temperature of the IPS glow. I had two Nano IPS devices myself for a brief moment and the terrible warm IPS glow was one of the main reasons why I sent them back.

**Comment by aufkrawall** [comment]
To share some experiences and thoughts after extended usage and calibrations:

- With an X-Rite i1 DisplayPro, I haven't seen the weird close-to-black tint issue ever again (wide gamut IPS displays, used 94% P3 Phosphor WLED spectral correction)
- Calibration step with gamma measurements in DisplayCal seem to contribute nothing that could aid novideo_srgb, verification results don't seem to be affected by it at all
- curves + matrix profile format seems to achieve better gamma curve results
- more than 1 testchart for each primary color doesn't help at all to achieve lower deviation, also not with XYZ LUT + matrix profile
- XYZ LUT + matrix profile format doesn't achieve lower color deviations vs. curves (unless a full 3D LUT is used with other programs)
- gamma curve validation results are already pretty solid with few grayscale patches in profiling test chart, deviations might be just different vs. VCGT (e.g. could be inverted, which would lead to noticeable brightness differences between VCGT or novideo_srgb, but both may be similarly correct or wrong)
- so gamma correction via VCGT medium speed in DisplayCal doesn't necessarily achieve better results
- not sure if there are meaningful improvements beyond e.g. 52 grayscale patches, as we would need to systematically increase the patch number and use a grayscale verification testchart that is maxed out (which is 256 grayscale patches) to make sure some improvements aren't just the flipped coin side of deteriorated others (which is my impression what mostly happens when increasing grayscale testchart number further, but may be different with different displays)
- there are differences in gamma curve dependent on the number of grayscale patches used during profiling, e.g. 52 scores less deviation than 256 during verification for darkest 5% measurement point of default verification test chart (0.03 vs. 0.06)
- DisplayCal has some minor bug/annoyance, the shown grayscale testchart number in the top field doesn't match the actual number used for e.g. default testchart for matrix profiles (it shows 52, but in the actual testchart table it's <30)
- according to verification results, novideo_srgb can achieve both rel. gamma 2.4 and BT.1886 well (also tested the latter with DisplayCal's graybalance verification testchart)
- side note: increased room/display/colorimeter temperature e.g. during heatwaves affects measurements, I then need to adjust RGB gain in the monitor's OSD differently to achieve 6500K (afair the X-Rite also features a note to be used with room temperature of 20°C)
- dithering options don't seem to affect measurements at all
- still zero issues with novideo_srgb after ~9 months of usage, including persistence of calibration etc. :)
- hard to imagine buying a graphics card with which I'd lose this program :( :)

**Comment by aufkrawall** [comment]
novideo_srgb (left) with 52 grayscale patches actually achieves better BT.1886 gamma curve accuracy than VCGT with medium calibration speed (right):
![gamma](https://user-images.githubusercontent.com/13510490/193661926-ab7b5682-718d-4136-9f93-c78fc4b10f55.png)

VCGT is noticeably darker with near blacks, but it's apparently wrong.

**Comment by ossnorry** [comment]
For correcting white point when monitor RGB isn't enough, so I still need to do a tone curve calibration or is the matrix enough? It seems like I'm getting closer to 6504k with tone curve, but I could be making a mistake with my test chart.

**Comment by erusapien** [comment]
sorry to revive and old thread, how do you create an synthetic profile? @James-F2 
I mean how to enter those values? Where can I obtain the values for my monitor to input into synth profile?
